eth/fetcher/tx_fetcher.go

```diff
@@ -148,7 +150,7 @@ type TxFetcher struct {
 	drop    chan *txDrop
 	quit    chan struct{}
 
-	underpriced mapset.Set[common.Hash] // Transactions discarded as too cheap (don't re-fetch)
+	underpriced map[common.Hash]int64 // Transactions discarded as too cheap (don't re-fetch)
```

Comment:
The mapset.Set is thread-safe, while map is not. This `underpriced` set will be read/write in different threads, e.g. `Notify` and `Enqueue`. Please either revert the change, or replace with a thread-safe struct.